tools/toolchain/gen_mksurf_namelist.py

```diff
@@ -0,0 +1,658 @@
+                    dest="input_path",
+                    default="/glade/p/cesm/cseg/inputdata/lnd/clm2/rawdata/")
+        parser.add_argument('-d','--debug', 
+                    help='Debug mode will print more information. ', 
+                    action="store_true", 
+                    dest="debug", 
+                    default=False)
+        parser.add_argument('-v','--verbose', 
+                    help='Verbose mode will print more information. ', 
+                    action="store_true", 
+                    dest="verbose", 
+                    default=False)
+        parser.add_argument('--vic', 
+                    help='''
+                    Add the fields required for the VIC model.
+                    [default: %(default)s]
+                    ''', 
+                    action="store_true", 
+                    dest="vic_flag", 
+                    default=False)
+        parser.add_argument('--glc', 
+                    help='''
+                    Add the optional 3D glacier fields for verification of the glacier model.
+                    [default: %(default)s]
+                    ''', 
+                    action="store_true", 
+                    dest="glc_flag", 
+                    default=False)
+        parser.add_argument('--hirespft', 
+                    help='''
+                    If you want to use the high-resolution pft dataset rather
+                    than the default lower resolution dataset.
+                    (Low resolution is at quarter-degree, high resolution at 3-minute)
+                    [Note: hires only available for 1850 and 2005.]
+                    ''', 
+                    action="store_true", 
+                    dest="hres_flag", 
+                    default=False)
+        parser.add_argument('--crop', 
+                    help='''
+                    Create datasets with the extensive list of prognostic crop types.
+                    [default: %(default)s]
+                    ''', 
+                    action="store",
+                    type = str2bool,  
+                    nargs = '?',
+                    const = True,
+                    required = False,
+                    dest="crop_flag", 
+                    default=True)
+        parser.add_argument('-f','--fast', 
+                    help='Toggle fast mode which does not user the large mapping file', 
+                    action="store_true", 
+                    dest="fast_flag", 
+                    default=False)
+        parser.add_argument('-r','--res',
+                    help='''
+                    Resolution is the supported resolution(s) to use for files.
+                    [default: %(default)s]
+                    ''',
+                    action="store", 
+                    dest="res", 
+                    choices=valid_opts['res'], 
+                    required=False, 
+                    default="4x5")
+        return parser
+
+
+def str2bool(v):
```

Comment:
This function needed only if the group prefers
--crop y/n

Reply:
Also this function already exists elsewhere:
https://github.com/negin513/ctsm/blob/b854ad6b0b3e393cf4f1ab100edd3d085ef0cd14/tools/contrib/subset_data.py#L326-L349

So better to make a single copy of this function somewhere. I don't know where the correct location would be.

Reply:
removed also the code is copied in utils.py.